sdk/cosmos/azure-cosmos/azure/cosmos/_execution_context/aio/base_execution_context.py

```diff
@@ -115,23 +114,29 @@ async def _fetch_items_helper_no_retries(self, fetch_function):
         :param Callable fetch_function: The function that fetches the items.
         :return: List of fetched items.
         :rtype: list
-        """
+        """ 
         fetched_items = []
-        # Continues pages till finds a non-empty page or all results are exhausted
         while self._continuation or not self._has_started:
```

Comment:
Originally in PR 35022 this was just `while True:` which resulted in some infinite loop scenarios.